.github/workflows/ci_pr.yml

```diff
@@ -0,0 +1,140 @@
+name: CI Unit tests
+
+on:
+  push:
+    branches: [ "*" ]
+  pull_request:
+    branches: [ "*" ]
+
+jobs:
+  legacy:
+    
+    name: "Python 2.6 Unit Tests"
+    runs-on: ubuntu-16.04
+
+    strategy:
+      fail-fast: false
+    env:
+      NOSEOPTS: "--verbose"
+
+    steps:
+
+    - name: Install Python 2.6
+      run: |
+        sudo add-apt-repository ppa:deadsnakes/ppa -y
+        sudo apt-get update -y
+        sudo apt install python2.6 python2.6-dev -y
+        sudo ln -sf /usr/bin/python2.6 /usr/bin/python
+        PATH="/usr/bin/${PATH:+:${PATH}}"
+        echo $PATH
+        echo "Installed python2.6"
+        python -V
+        curl https://bootstrap.pypa.io/2.6/get-pip.py -o get-pip.py
+        sudo env "PATH=$PATH" python get-pip.py
+
+    - uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: |
+        sudo env "PATH=$PATH" pip install -r requirements.txt
+        sudo env "PATH=$PATH" pip install -r test-requirements.txt
+    
+    - name: Test with nosetests
+      run: |
+        ./ci/nosetests.sh || EXIT_CODE=$?
+        exit $EXIT_CODE
+
+  build:
+    
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+
+          - python-version: 2.7
+            PYLINTOPTS: "--rcfile=ci/2.7.pylintrc"
+            NOSEOPTS: "--verbose --with-timer"
+            PYLINTFILES: "azurelinuxagent setup.py makepkg.py tests"
```

Comment:
nit: Is it possible to extract these flags into variables so we don't repeat their values for each python job?

Reply:
What precisely are you referring to? Like using a default value that gets overridden, e.g. `--rcfile=ci/3.6.pylintrc` as a default that can be overridden for python2.7 and 3.4?

Reply:
That's what I was thinking, but I see we do the exact same thing in Travis, so you can ignore for now